alot/buffers.py

```diff
@@ -367,14 +368,29 @@ def rebuild(self):
 
         self._tree = ThreadTree(self.thread)
 
-        bars_att = settings.get_theming_attribute('thread', 'arrow_bars')
-        heads_att = settings.get_theming_attribute('thread', 'arrow_heads')
-        A = ArrowTree(
-            self._tree,
-            indent=2,
-            childbar_offset=0,
-            arrow_tip_att=heads_att,
-            arrow_att=bars_att)
+        # define A to be the tree to be wrapped by a NestedTree and displayed.
+        # We wrap the thread tree into an ArrowTree for decoration if
+        # indentation was requested and otherwise use it as is.
+        if self._indent_width == 0:
+            A = self._tree
+        else:
+            # we want decoration.
+            bars_att = settings.get_theming_attribute('thread', 'arrow_bars')
+            # only add arrow heads if there is space (indent > 1).
+            heads_char = None
+            heads_att = None
+            if self._indent_width > 1:
+                heads_char = u'\u27a4'
```

Comment:
feel free to take this or leave this, but if we added `#encoding = utf-8` to the top of the file we could insert the unicode character instead of the escape code here.

Reply:
yes, right. I'll do that.

Reply:
Just tested this, and it triggers a weird error related to urwid not being able to determine the correct width..Or rather, demanding more than width 1 for 

```
  File "build/bdist.linux-x86_64/egg/urwidtrees/widgets.py", line 68, in _get
  File "build/bdist.linux-x86_64/egg/urwidtrees/lru_cache.py", line 102, in wrapper
  File "build/bdist.linux-x86_64/egg/urwidtrees/widgets.py", line 54, in __getitem__
  File "build/bdist.linux-x86_64/egg/urwidtrees/nested.py", line 88, in get_decorated
  File "build/bdist.linux-x86_64/egg/urwidtrees/nested.py", line 84, in _get_decorated_entry
  File "build/bdist.linux-x86_64/egg/urwidtrees/decoration.py", line 446, in decorate
  File "build/bdist.linux-x86_64/egg/urwidtrees/decoration.py", line 416, in _construct_first_indent
urwidtrees.decoration.NoSpaceError
```

```patch
--- a/alot/buffers.py
+++ b/alot/buffers.py
@@ -1,3 +1,4 @@
+# encoding=utf-8
 # Copyright (C) 2011-2012  Patrick Totzke <patricktotzke@gmail.com>
 # This file is released under the GNU GPL, version 3 or a later revision.
 # For further details see the COPYING file
@@ -380,7 +381,7 @@ class ThreadBuffer(Buffer):
             heads_char = None
             heads_att = None
             if self._indent_width > 1:
-                heads_char = u'\u27a4'
+                heads_char = '➤'
                 heads_att = settings.get_theming_attribute('thread',
                                                            'arrow_heads')
             A = ArrowTree(

```
```

Reply:
Then let's not worry about it for now. unicode support in the python2 world is often less than ideal, as we've seen lately.